File: q1015/morningquest2.c
#include <stdio.h>
int main(){
	const int mod=1000000007;
	const int target=100000000;
	int i=3,a=1,b=1,c;
	for(;i<=target;i++){
		c=(a+b)%mod;
		a=b;
		b=c;
	}
	printf("%d\n",c);
	return 0;
}

File: q1015/morningquest2.rb
#!/usr/bin/ruby
MOD=1000000007
g=Enumerator.new{|y|
	a=1
	b=1
	y<<a
	loop{
		y<<b
		b,a=(a+b)%MOD,b
	}
}
p g.lazy.drop(100000000-1).take(1).first